src/navigation/orbitalnavigator.cpp

```diff
@@ -1083,45 +1083,39 @@ glm::dvec3 OrbitalNavigator::cameraToSurfaceVector(const glm::dvec3& cameraPos,
     return centerToActualSurface - posDiff;
 }
 
-void OrbitalNavigator::setFocusNode(const SceneGraphNode* focusNode,
-                                    bool resetVelocitiesOnChange)
-{
-    const SceneGraphNode* previousAnchor = _anchorNode;
-    setAnchorNode(focusNode, resetVelocitiesOnChange);
-    setAimNode(nullptr);
-
-    global::eventEngine->publishEvent<events::EventFocusNodeChanged>(
-        previousAnchor,
-        focusNode
-    );
+void OrbitalNavigator::setFocusNode(const SceneGraphNode* node, bool resetVelocities) {
+    setFocusNode(node->identifier(), resetVelocities);
 }
 
-void OrbitalNavigator::setFocusNode(const std::string& focusNode, bool) {
-    _anchor = focusNode;
+void OrbitalNavigator::setFocusNode(const std::string& identifier, bool resetVelocities) {
+    _resetVelocitiesOnAnchorChange = resetVelocities;
+    _anchor = identifier;
     _aim = std::string("");
```

Comment:
```suggestion
    _aim = std::string();
```

While you are already in here?